src/pybind/mgr/cephadm/agent.py

```diff
@@ -130,22 +146,112 @@ def shutdown(self) -> None:
         self.cherrypy_shutdown_event.set()
 
 
-class Root:
-    exposed = True
+class Root(object):
+
+    # collapse everything to '/'
+    def _cp_dispatch(self, vpath: str) -> 'Root':
+        cherrypy.request.path = ''
+        return self
 
     def __init__(self, mgr: "CephadmOrchestrator"):
         self.mgr = mgr
-        self.data = HostData(self.mgr)
 
-    def GET(self) -> str:
+    @cherrypy.expose
+    def index(self) -> str:
         return '''<!DOCTYPE html>
 <html>
 <head><title>Cephadm HTTP Endpoint</title></head>
 <body>
-<p>Cephadm HTTP Endpoint is up and running</p>
+<p>Cephadm Service discovery Endpoints</p>
+<p><a href='sd-config?service=prometheus'>Prometheus http sd-config</a></p>
```

Comment:
this isn't that clear - since the prometheus service is actually the prometheus server. What this is referring to is the mgr/prometheus module - so perhaps change the test to reflect that?

Reply:
I changed Prometheus by mgr/Prometheus

Reply:
Thanks. Not sure about the service=prometheus bit. It's probably a nit-pick, but the service is mgr-prometheus - the sd_config doesn't represent the endpoints of prometheus itself. I also left a comment about alertmanager. I think that needs a little more discussion too.

Reply:
I see your point. What about **service=mgr-prometheus**?

Reply:
works for me

Reply:
Fixed